action/protocol/poll/candidateindexer_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package poll
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/iotexproject/iotex-core/action/protocol/vote"
+	"github.com/iotexproject/iotex-core/db"
+	"github.com/iotexproject/iotex-core/state"
+	"github.com/iotexproject/iotex-core/test/identityset"
+)
+
+func TestCandidateIndexer(t *testing.T) {
+	require := require.New(t)
+	indexer, err := NewCandidateIndexer(db.NewMemKVStore())
+	require.NoError(err)
+	require.NoError(indexer.Start(context.Background()))
+	candidates := state.CandidateList{
+		{
+			Address:       identityset.Address(1).String(),
+			Votes:         big.NewInt(30),
+			RewardAddress: "rewardAddress1",
+		},
+		{
+			Address:       identityset.Address(2).String(),
+			Votes:         big.NewInt(22),
+			RewardAddress: "rewardAddress2",
+		},
+		{
+			Address:       identityset.Address(3).String(),
+			Votes:         big.NewInt(20),
+			RewardAddress: "rewardAddress3",
+		},
+		{
+			Address:       identityset.Address(4).String(),
+			Votes:         big.NewInt(10),
+			RewardAddress: "rewardAddress4",
+		},
+	}
+	require.NoError(indexer.PutCandidateList(uint64(1), &candidates))
+	candidates2, err := indexer.CandidateList(uint64(1))
+	require.NoError(err)
+	for i, cand := range candidates {
+		require.True(cand.Equal(candidates2[i]))
+	}
```

Comment:
even through the data reading from the indexer can not have extra fields, but we still need to check
```require.Equal(len(candidates2), len(candidates))```